crates/ruff_linter/src/rules/pyupgrade/rules/use_pep604_annotation.rs

```diff
@@ -209,5 +209,6 @@ fn is_allowed_value(expr: &Expr) -> bool {
         | Expr::Starred(_)
         | Expr::Slice(_)
         | Expr::IpyEscapeCommand(_) => false,
+        Expr::StringList(_) => todo!(),
```

Comment:
We should pick a different name than string list (`StringConcatenation` maybe?), because string list sounds like `["1", "2"]` rather than `"1" "2"`

Reply:
Yeah, thanks! I'll propose a few names, we can finalize then.